### ISSUE TITLE
HighlightFieldDictionary constructor creates a shallow copy object

### ISSUE THREAD

**Issue by techninja24** [elastic/elasticsearch-net#3206]
**NEST/Elasticsearch.Net version**: latest

**Elasticsearch version**: latest

**Description of the problem including expected versus actual behavior**:

public HighlightFieldDictionary(IDictionary<string, HighlightHit> dictionary = null)
		{
			if (dictionary == null) return;
			foreach(var kv in dictionary)
			{
				this.Add(kv.Key, kv.Value);
			}
		}
In the above code kv.Value is of type HighlightHit which is a reference type. So this constructor is doing a shallow copy of the object. It should be deep copy.

**Comment by techninja24** [comment]
Hi @Mpdreamz Do you think this is a valid issue? I faced a problem while writing tests against this where I wanted a copy before sending it to a function which does some changes in the dictionary.

**Comment by techninja24** [comment]
I have repeated highlights for fields analyzed with different analyzers. I have a utility function say Modify() which does some processing on these highlight hits like pruning duplicates, selecting largest highlight etc. For the purpose of unit testing this function I wanted to create a copy of the original object to verify the changes made in the Modify() function. 
This is when I discovered that the object is a shallow copy since any changes I made would always reflect in the copy object. 

This was some test code which even I am not using now. But still there could be cases where someone will want a deep copy for similar cases mostly I think is for testing purposes.
